LCLSGeneral/LCLSGeneral/POUs/Data/FB_DataBuffer.TcPOU

```diff
@@ -0,0 +1,55 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.18">
+  <POU Name="FB_DataBuffer" Id="{663bb4ee-ee74-4dfb-9a41-308201d6b012}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_DataBuffer
+(*
+	Function Block to accumulate data into an array.
+	2019-10-09 Zachary Lentz
+	
+	Requires the user to supply pointers to the value and to 2 arrays:
+	1. A partial buffer that we will slowly fill one value at a time
+	2. An output buffer that will only update when the partial buffer is full
+	
+	Take great care of the following, or else your program will likely crash,
+	or at least have corrupt data:
+	1. The input type and array types must match
+	2. The provided element count must be accurate and match both arrays
+	3. The provided element size is correct
+	
+	As this function block as no way of checking that you did this correctly.
```

Comment:
In general when dealing with references in PLC code, a good practice is to always call the function block with these inputs updated with, ADR(). If an FB is instantiated once with ADR(Array) as an input, then there is always a chance that an online change will shift the address of that array and the address will become invalid. I've made this mistake a few times, and it's a hard crash, made worse if you enabled auto start.

Reply:
Should I include this as part of the docs for the function block?

Reply:
That's scary...

Reply:
> Should I include this as part of the docs for the function block?

If you like, but it's one of those best practices that applies to so many things. I just wanted to mention it.

Reply:
Ok, was just asking because you submitted it as a code review comment. I may opt to put it on confluence instead on a twincat best-practices page.